rgo/builders/copr.py

```diff
@@ -53,39 +53,44 @@ def __init__(
         for chroot in chroots:
             if chroot not in self.client.mock_chroot_proxy.get_list():
                 raise Exception("{!r} is not an active chroot".format(chroot))
-        self.chroots = chroots
+        self.cmdline_input_chroots = set(chroots)
```

Comment:
Unless I'm missing something, I see this distinction of `cmdline_input_chroots` and `all_project_chroots` as an unnecessary complication. `self.chroots` simply represented the chroots you're building, which is the only thing you care about after figuring out this set, which is either what came from the CLI, or what is set on the project. No need to store both and then add another `if` when you're going to use the values in `build`.

The only difference seems to be `self.chroots` weren't set if they didn't come from CLI, you can set them to the project chroots for that case and the rest of the code can work with this one value.

Reply:
Yes you are right!
Thanks.